core/controllers/learner_group.py

```diff
@@ -1006,7 +1144,22 @@ def get(self) -> None:
         })
 
 
-class LearnerGroupProgressSharingPermissionHandler(base.BaseHandler):
+class LearnerGroupProgressSharingPermissionHandlerNormalizedPayloadDict(
+    TypedDict
+):
+    """Dict representation of LearnerGroupProgressSharingPermissionHandler's
+    normalized_payload dictionary.
+    """
+
+    progress_sharing_permission: Union[str, bool]
```

Comment:
Ditto as above. This should only be `str`.

Reply:
ditto as above

Reply:
Todo is added #16566